### ISSUE TITLE
Problem with BTProgressHUD after upgrading to unified API

### ISSUE THREAD

**Issue by robertbird** [aritchie/acr-xamarin-forms#76]
Hi,

I've upgraded my iOS project to the unified API and also upgraded the Acr.XamForms.UserDialogs nuget package to version 3.0.0

I am now seeing a null reference error which appears to be caused by the BTProgressHUD. (Please see the stacktrace below)

I have tried uninstalling and re-installing the Acr nuget package. Then I tried removing the BTProgressHud dll reference and manually adding the latest version (via components) instead. But still no luck.

Has anyone else seen this? Any suggestions?

Thanks,
Rob

Stacktrace:

    Unhandled Exception:
    
    System.NullReferenceException: Object reference not set to an instance of an object
    
    2015-02-05 00:38:16.382 PulseMobileContractoriOS[3515:1200247] Unhandled managed exception: Object reference not set to an instance of an object (System.NullReferenceException)
      at BigTed.ProgressHUD.PositionHUD (Foundation.NSNotification notification) [0x00000] in <filename unknown>:0 
      at BigTed.ProgressHUD.ShowProgressWorker (Single progress, System.String status, MaskType maskType, Boolean textOnly, ToastPosition toastPosition, System.String cancelCaption, System.Action cancelCallback, Double timeoutMs, Boolean showContinuousProgress, UIKit.UIImage displayContinuousImage) [0x00000] in <filename unknown>:0 
      at BigTed.ProgressHUD+<Show>c__AnonStorey0.<>m__0 () [0x00000] in <filename unknown>:0 
      at Foundation.NSActionDispatcher.Apply () [0x00000] in /Developer/MonoTouch/Source/maccore/src/Foundation/NSAction.cs:56 
      at (wrapper managed-to-native) ObjCRuntime.Messaging:void_objc_msgSend_IntPtr_IntPtr_bool (intptr,intptr,intptr,intptr,bool)
      at Foundation.NSObject.InvokeOnMainThread (System.Action acti
    on) [0x00007] in /Developer/MonoTouch/Source/maccore/src/Foundation/NSObject2.cs:575 
      at BigTed.ProgressHUD.Show (System.String status, Single progress, MaskType maskType, Double timeoutMs) [0x00000] in <filename unknown>:0 
      at BigTed.BTProgressHUD.Show (System.String status, Single progress, MaskType maskType) [0x00000] in <filename unknown>:0 
      at Acr.XamForms.UserDialogs.iOS.ProgressDialog.Refresh () [0x00000] in <filename unknown>:0 
      at Acr.XamForms.UserDialogs.iOS.ProgressDialog.Show () [0x00000] in <filename unknown>:0 
      at Acr.XamForms.UserDialogs.AbstractUserDialogService.Progress (Acr.XamForms.UserDialogs.ProgressConfig config) [0x00000] in <filename unknown>:0 
      at Acr.XamForms.UserDialogs.AbstractUserDialogService.Progress (System.String title, System.Action onCancel, System.String cancelText, Boolean show) [0x00000] in <filename unknown>:0 
      at Pulse.Mobile.Core.ViewModel.ViewModelBase.GetProgress (System.String message) [0x00001] in d:\Dev\TFS2010\Adactus.Pulse\Mobile\POC\Main-UnifiedAP
    I\Solutions\Core\Core.ViewModel\ViewModelBase.cs:60 
      at Pulse.Mobile.Core.ViewModels.Diary.DiaryViewModel+<LoadAppointments>d__11.MoveNext () [0x00038] in d:\Dev\TFS2010\Adactus.Pulse\Mobile\POC\Main-UnifiedAPI\Solutions\Core\Core.ViewModel\Diary\DiaryViewModel.cs:151 
    --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000b] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.ExceptionServices/ExceptionDispatchInfo.cs:62 
      at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00034] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.CompilerServices/TaskAwaiter.cs:60 
      at Pulse.Mobile.Core.ViewModel.Framework.PulseMobileCommand+<EnsureAuthenticated>d__d.MoveNext () [0x00059] in d:\Dev\TFS2010\Adactus.Pulse\Mobile\POC\Main-UnifiedAPI\Solutions\Core\Core.ViewModel\Framework\PulseMobileCommand.cs:166 
    --- End of stack trace from previous location where exception was thr
    own ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000b] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.ExceptionServices/ExceptionDispatchInfo.cs:62 
      at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00034] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.CompilerServices/TaskAwaiter.cs:60 
      at Pulse.Mobile.Core.ViewModel.Framework.PulseMobileCommand+<EnsureAuthenticated>d__d.MoveNext () [0x00189] in d:\Dev\TFS2010\Adactus.Pulse\Mobile\POC\Main-UnifiedAPI\Solutions\Core\Core.ViewModel\Framework\PulseMobileCommand.cs:187 
    --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000b] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.ExceptionServices/ExceptionDispatchInfo.cs:62 
      at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00034] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime
    .CompilerServices/TaskAwaiter.cs:60 
      at Pulse.Mobile.Core.ViewModel.Framework.PulseMobileCommand+<EnsureAuthenticated>d__d.MoveNext () [0x00189] in d:\Dev\TFS2010\Adactus.Pulse\Mobile\POC\Main-UnifiedAPI\Solutions\Core\Core.ViewModel\Framework\PulseMobileCommand.cs:187 
    --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000b] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.ExceptionServices/ExceptionDispatchInfo.cs:62 
      at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00034] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.CompilerServices/TaskAwaiter.cs:60 
      at Pulse.Mobile.Core.ViewModel.Framework.PulseMobileCommand+<EnsureAuthenticated>d__d.MoveNext () [0x00189] in d:\Dev\TFS2010\Adactus.Pulse\Mobile\POC\Main-UnifiedAPI\Solutions\Core\Core.ViewModel\Framework\PulseMobileCommand.cs:187 
    --- End of stack trace from previous location where exception was th
    rown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000b] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.ExceptionServices/ExceptionDispatchInfo.cs:62 
      at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00034] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.CompilerServices/TaskAwaiter.cs:60 
      at Pulse.Mobile.Core.ViewModel.Framework.PulseMobileCommand+<Execute>d__a.MoveNext () [0x0001e] in d:\Dev\TFS2010\Adactus.Pulse\Mobile\POC\Main-UnifiedAPI\Solutions\Core\Core.ViewModel\Framework\PulseMobileCommand.cs:123 
    2015-02-05 00:38:16.383 PulseMobileContractoriOS[3515:1200247]

**Comment by aritchie** [comment]
The issue doesn't look like anything caused by the unified update.  I'm probably going to need some code to get any sense of what the issue could be

**Comment by robertbird** [comment]
Thanks for the update, I will take a look after work tonight and see if I can find the exact line that causes the problem and maybe create some sort of sample to illustrate.

**Comment by robertbird** [comment]
I have been having so much trouble with this, but am starting to think you are right that it is my definitely my code to blame. 

The strange thing is that it used to work before the update to the unified API. However I am using a few libraries and they have had to be upgraded too, so there are a lot of changes going on.

I will have one last attempt tonight and then close this ticket one way or another as I'm sure its something stupid I am doing.

**Comment by robertbird** [comment]
Ok so a bit of progress... 

One problem I found was to do with my use of XLabs and use of their XFormsAppiOS class when launching the iOS app. I have pulled this out and it seems much healthier. 

I also think the other screen where the app was crashing is because I seem to be showing the loading spinner before the screen has actually been displayed. i.e. the X.F. page has been created and in the BindingContext changed event I fire off a command in my viewModel to load data. But this is all happening before the page has actually been displayed using the navigationPage.

Again, the strange thing is that this was working exactly the same way before. I'm thinking that maybe the new Application class in X.F 1.3 has changed this behaviour somehow.
